pkg/scalers/forgejo_runner_scaler.go

```diff
@@ -0,0 +1,235 @@
+	client     *http.Client
+	logger     logr.Logger
+}
+
+type forgejoRunnerMetadata struct {
+	triggerIndex int
+
+	Token   string `keda:"name=token, order=authParams;triggerMetadata"`
+	Address string `keda:"name=address, order=triggerMetadata"`
+	Labels  string `keda:"name=labels, order=triggerMetadata"` // comma separated runner labels
+	Global  bool   `keda:"name=global, order=triggerMetadata, optional"`
+	Owner   string `keda:"name=owner, order=triggerMetadata, optional"`
+	Org     string `keda:"name=org, order=triggerMetadata, optional"`
+	Repo    string `keda:"name=repo, order=triggerMetadata, optional"`
+}
+
+func parseForgejoRunnerMetadata(config *scalersconfig.ScalerConfig) (*forgejoRunnerMetadata, error) {
+	meta := &forgejoRunnerMetadata{}
+	meta.triggerIndex = config.TriggerIndex
+
+	if err := config.TypedConfig(meta); err != nil {
+		return nil, fmt.Errorf("error parsing forgejo metadata: %w", err)
+	}
+
+	if meta.Address[len(meta.Address)-1:] == "/" {
+		meta.Address = meta.Address[:len(meta.Address)-1]
+	}
+
+	return meta, nil
+}
+
+// NewForgejoRunnerScaler creates a new Forgejo Runner Scaler
+func NewForgejoRunnerScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	c := kedautil.CreateHTTPClient(30*time.Second, false)
+
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %w", err)
+	}
+
+	meta, err := parseForgejoRunnerMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Forgejo Runner metadata: %w", err)
+	}
+
+	logger := InitializeLogger(config, "forgejo_runner_scaler")
+
+	return &forgejoRunnerScaler{
+		client:     c,
+		metricType: metricType,
+		metadata:   meta,
+		logger:     logger,
+	}, nil
+}
+
+func (s *forgejoRunnerScaler) GetMetricsAndActivity(ctx context.Context, metricName string) ([]external_metrics.ExternalMetricValue, bool, error) {
+	jobList, err := s.getJobsList(ctx)
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, false, err
+	}
+
+	metric := GenerateMetricInMili(metricName, float64(len(jobList)))
+
+	metric.Value.Add(resource.Quantity{})
+
+	return []external_metrics.ExternalMetricValue{metric}, true, nil
+}
+
+func (s *forgejoRunnerScaler) getJobsList(ctx context.Context) ([]ForgejoJob, error) {
+	var jobList []ForgejoJob
+
+	uri, err := s.getRunnerJobURL()
+	if err != nil {
+		return jobList, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "GET", uri.String(), nil)
+	if err != nil {
+		return jobList, err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("token %s", s.metadata.Token))
+
+	r, err := s.client.Do(req)
+	if err != nil {
+		return jobList, err
+	}
+
+	b, err := io.ReadAll(r.Body)
+	if err != nil {
+		return jobList, err
+	}
+	_ = r.Body.Close()
+
+	if r.StatusCode != 200 {
+		return jobList,
+			fmt.Errorf("the Forgejo REST API returned error. url: %s status: %d response: %s",
+				s.metadata.Address,
+				r.StatusCode,
+				string(b),
+			)
+	}
```

Comment:
I think we must use defer immediately after getting response, because if the io.ReadAll() fails the body never gets closed.

```suggestion
	defer func() {
        if closeErr := r.Body.Close(); closeErr != nil {
            s.logger.Error(closeErr, "Failed to close response body")
        }
    }()

    if r.StatusCode != 200 {
        b, _ := io.ReadAll(r.Body)
        return jobList, fmt.Errorf("the Forgejo REST API returned error. url: %s status: %d response: %s",
            s.metadata.Address,
            r.StatusCode,
            string(b),
        )
    }

    b, err := io.ReadAll(r.Body)
    if err != nil {
        return jobList, err
    }
```